### PR TITLE
[pugixml] update to 1.12.1

### PR DESCRIPTION
**Pull request by gamecat10**
**Describe the pull request**

- #### What does your PR fix?
  - update pugixml to 1.12.1
  - add new feature "wchar-mode"

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/pugixml/portfile.cmake [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
 )
 
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+  FEATURES
+    wchar-mode	PUGIXML_WCHAR_MODE
+)
+
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/pugixml/portfile.cmake:16]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/pugixml/portfile.cmake [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
-    OPTIONS -DUSE_POSTFIX=ON
+    OPTIONS -DPUGIXML_USE_POSTFIX=ON
+      ${FEATURE_OPTIONS}
 )
 
 vcpkg_install_cmake()
```
Comment (JonLiu1993) [inline ports/pugixml/portfile.cmake:21]:
```suggestion
vcpkg_cmake_install()
```

File: ports/pugixml/portfile.cmake [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
 vcpkg_copy_pdbs()
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/pugixml)
```
Comment (JonLiu1993) [inline ports/pugixml/portfile.cmake:23]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/pugixml)
```

File: ports/pugixml/portfile.cmake [hunk 4]
```diff
@@ -1,22 +1,32 @@
...
 vcpkg_fixup_pkgconfig()
 
+if ("wchar-mode" IN_LIST FEATURES)
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/pugiconfig.hpp" "// #define PUGIXML_WCHAR_MODE" "#define PUGIXML_WCHAR_MODE")
+endif()
+
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
```
Comment (JonLiu1993) [inline ports/pugixml/portfile.cmake:30]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static
- x64-linux

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
You'll need to either make `PUGIXML_USE_WCHAR` a triplet variable, or do the suggested solution here https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#recommended-solutions

**Comment by JonLiu1993** [comment]
@gamecat10, please fix the conflicting files, thanks!

**Comment by JonLiu1993** [comment]
Draft this PR until it has any progress.

**Comment by JonLiu1993** [comment]
Closing this PR, please reopen if you continue this work.

**Pull request closed without merging**